platform/posix/transport/src/plaintext_posix.c

```diff
@@ -113,30 +69,72 @@ int32_t Plaintext_Recv( const NetworkContext_t * pNetworkContext,
                         void * pBuffer,
                         size_t bytesToRecv )
 {
-    int32_t bytesReceived = 0;
+    int32_t bytesReceived = -1, selectStatus = -1, getTimeoutStatus = -1;
+    struct timeval transportTimeout, selectTimeout;
+    socklen_t transportTimeoutLen;
+    fd_set readfds;
 
-    bytesReceived = ( int32_t ) recv( pNetworkContext->socketDescriptor, pBuffer, bytesToRecv, 0 );
+    assert( pNetworkContext != NULL );
+    assert( pBuffer != NULL );
+
+    FD_ZERO( &readfds );
+    FD_SET( pNetworkContext->socketDescriptor, &readfds );
+
+    getTimeoutStatus = getsockopt( pNetworkContext->socketDescriptor,
+                                   SOL_SOCKET,
+                                   SO_RCVTIMEO,
+                                   &transportTimeout,
+                                   &transportTimeoutLen );
+
+    if( getTimeoutStatus < 0 )
+    {
+        /* Make #select return immediately if getting the timeout failed. */
+        selectTimeout.tv_sec = 0;
+        selectTimeout.tv_usec = 0;
+    }
+    else
+    {
+        selectTimeout.tv_sec = transportTimeout.tv_sec;
+        selectTimeout.tv_usec = transportTimeout.tv_usec;
+    }
```

Comment:
can we use the transportTimeout itself? Do we need selectTimeout?

Reply:
Removing `selectTimeout`